### ISSUE TITLE
Consider adding name of server config to SSHOptions

### ISSUE THREAD

**Issue by pedromss** [shmishleniy/sbt-deploy-ssh#7]
I have several config files each suffixed with the environment's name:

- application.conf // for running locally
- application.qa.conf // for QA environment
- application.,test.conf // for testing enviroment
- application.prod.conf // for production

Now I would like to use `scp` in `deploySshExecBefore` to copy these files depending on the environment I'm deploying to. Currently I'm forced to do:

```scala
lazy val testConfig = ServerConfig(name = "test", /* ..., */)
lazy val qaConfig = ServerConfig(name = "qa", /* ..., */)
lazy val prodConfig = ServerConfig(name = "prod", /* ..., */)

val currentConfigHost = ssh.options.host
val currentConfigName = currentConfigHost match {
   case testConfig.host => testConfig.name
    case qaConfig.host => qaConfig.name
    case prodConfig.host => prodConfig.name
}

ssh.scp(proc => proc.send(file(s"./src/main/resources/application-$currentConfigName.conf"), s"${configDirectory.value}application.conf"))
```

I would like to be able to just do;

```scala
ssh.scp(proc => proc.send(file(s"./src/main/resources/application-${ssh.options.name.get}.conf"), s"${configDirectory.value}application.conf"))
```

**Comment by shmishleniy** [comment]
Hi,

You can rename your file for example `application.qa.conf` to `application.127.0.0.1.conf` (your server ip).

or use hostname instead of ip
servers = [
 {
  name = "server_1"
  host = "qa-hostname-env"
  user = "test"
 }
]

and then your file `applciation.qa-hostname-env.conf`

in these cases your code will be fully working:
`ssh.scp(proc => proc.send(file(s"./src/main/resources/application-${ssh.options.name.get}.conf"), s"${configDirectory.value}application.conf"))`

**Comment by pedromss** [comment]
That has the down side that my configuration file would be tied to the machine's hostname. If I changed the remote machine's address I would have to remember to rename the configuration files. Doesn't seem all that appealing to me.

I suggested this because I saw that you do this:

```scala
implicit val ssh = SSH(
      SSHOptions(serverConfig.host,
        serverConfig.user getOrElse Properties.userName,
        serverConfig.password,
        serverConfig.passphrase,
        port = serverConfig.port.getOrElse(22),
        identities = SSHIdentity(sshKey)::Nil)
    )
```

All it would take was for you to do this (line 151 of `DeploySSH.scala`:

```scala
implicit val ssh = SSH(
      SSHOptions(serverConfig.host,
        serverConfig.user getOrElse Properties.userName,
        serverConfig.password,
        serverConfig.passphrase,
        name = serverConfig.name, // <-- notice this line. SSHOptions already has a constructor parameter for the name.
        port = serverConfig.port.getOrElse(22),
        identities = SSHIdentity(sshKey)::Nil)
    )
```

**Comment by shmishleniy** [comment]
Ok, will be added to 1.3

**Comment by shmishleniy** [comment]
fixed in 0.1.3
